source/text/WrappedText.cpp

```diff
@@ -294,6 +298,10 @@ void WrappedText::Wrap()
 		word.x += width;
 		lineWidth = word.x;
 	}
+	// If we are not in a word, advance to the next line if this line has any words.
```

Comment:
Took me a few minutes of reading through this code to figure out how this fixed the bug. I think a slight change to this comment would help make it more clear what's happening.
```suggestion
	// If the text ended in whitespace but the line did contain a word at some point,
	// then we must still advance to the next line.
```
Maybe also update the comment `// Add this final word to the existing words.` a few lines above this to read `// Add this final word to the existing words and advance to the next line.`, using the "advance to the next line" vocab in both places to make a connection between what normally happens and this special case.

Reply:
You're not wrong - this section of code could use a bit more clarity.  I still get thrown by AdjustLine being the thing that sets lineBegin and lineWidth.